bpf/bpf_sock.c

```diff
@@ -332,12 +332,19 @@ static __always_inline int __sock4_xlate_fwd(struct bpf_sock_addr *ctx,
 	if (svc->affinity && !backend_from_affinity)
 		lb4_update_affinity_by_netns(svc, &id, backend_id);
 
+#if !defined(ENABLE_HOST_SERVICES_PEER)
+	/* revnat entry is not required for TCP protocol when getpeername mapping isn't enabled */
+	if (!udp_only && ctx->protocol == IPPROTO_TCP)
+		goto update_dst;
+#endif
```

Comment:
Good catch, thanks for the fix. Could you instead make `sock4_update_revnat()` and `sock6_update_revnat()` just `return 0` instead of `return -1`? That would achieve the same while not adding more ifdefs to the code.

Reply:
Thanks @borkmann , good idea; I just did that (and verified it still works).

Reply:
Awesome, thanks!